src/dune_console/dune_console.ml

```diff
@@ -292,7 +292,12 @@ module Backend = struct
           if elapsed >= frame_rate then last := now
           else
             let delta = frame_rate -. elapsed in
-            Unix.sleepf delta;
+            let (_
+                  : Unix.file_descr list
+                    * Unix.file_descr list
+                    * Unix.file_descr list) =
+              Unix.select [ Unix.stdin ] [] [] delta
```

Comment:
You shouldn't ignore the result. You should make sure that the first list contains an Fd.

* If it contains an Fd, you should handle pending events (if there are any)
* Otherwise, select returned because of the timeout and there's no need to check for events.

Reply:
OK, I've added some logic handling this now. I've also split the keypress handling into a separate function from render.